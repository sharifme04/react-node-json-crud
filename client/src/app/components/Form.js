import React from "react";
import PropTypes from "prop-types";

const Form = ({ handleChange, values, handleSubmit, btnColor, btnTitle }) => {
  return (
    <form onSubmit={handleSubmit}>
      <div className="form-group">
        <label>Title:</label>
        <input
          type="label"
          value={values?.title}
          onChange={handleChange}
          className="form-control"
          name="title"
          required
        />
      </div>
      <div className="form-group mt-3">
        <label>Description:</label>
        <input
          type="label"
          value={values?.description}
          onChange={handleChange}
          className="form-control"
          name="description"
          required
        />
      </div>
      <div className="mt-4"></div>
      <input
        type="submit"
        className={btnColor || "btn btn-primary"}
        value={btnTitle || "Add"}
      />
    </form>
  );
};

Form.propTypes = {
  handleSubmit: PropTypes.func,
  value: PropTypes.any,
  handleChange: PropTypes.func,
  btnColor: PropTypes.any,
  btnTitle: PropTypes.string,
};

export default Form;
