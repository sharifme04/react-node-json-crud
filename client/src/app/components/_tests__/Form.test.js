import React from "react";
import renderer from "react-test-renderer";
import Form from "../Form";

const values = { title: "sharif", description: "some description now" };

test("snapshot Form renders with empty props", () => {
  const component = renderer.create(<Form />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshot Form renders with props", () => {
  const component = renderer.create(
    <Form
      btnColor="btn btn-info"
      btnTitle="Update"
      values={values}
      handleChange={() => {}}
      handleSubmit={() => {}}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
