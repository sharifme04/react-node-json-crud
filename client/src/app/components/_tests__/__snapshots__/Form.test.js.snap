// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Form renders with empty props 1`] = `
<form>
  <div
    className="form-group"
  >
    <label>
      Title:
    </label>
    <input
      className="form-control"
      name="title"
      type="label"
    />
  </div>
  <div
    className="form-group mt-3"
  >
    <label>
      Description:
    </label>
    <input
      className="form-control"
      name="description"
      type="label"
    />
  </div>
  <div
    className="mt-4"
  />
  <input
    className="btn btn-primary"
    type="submit"
    value="Add"
  />
</form>
`;

exports[`snapshot Form renders with props 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label>
      Title:
    </label>
    <input
      className="form-control"
      name="title"
      onChange={[Function]}
      type="label"
      value="sharif"
    />
  </div>
  <div
    className="form-group mt-3"
  >
    <label>
      Description:
    </label>
    <input
      className="form-control"
      name="description"
      onChange={[Function]}
      type="label"
      value="some description now"
    />
  </div>
  <div
    className="mt-4"
  />
  <input
    className="btn btn-info"
    type="submit"
    value="Update"
  />
</form>
`;
